### DISCUSSION TITLE
Disclaimer im CMS über Sonstiges - User Interface - Linksammlung erstellen

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#62] [Q&A]
In WebGIS4 gibt es die Möglichkeit einen Disclaimer (Bestätigung eines Textes und Weiterleitung zur Karte) im CMS über Sonstiges - User Interface - Linksammlung zu erstellen.
Soll/Kann diese Funktion in WebGIS5 weiterverwendet werden?
Gibt es eine andere Möglichkeit nach Bestätigung eines Textes zum gewünschten Inhalt weiterzuleiten?

**Comment by gis-eni** [comment]
Nein, standardmäßig wird kein Disclaimer angeboten. Nicht jeder ruft die Karten gleich auf (WebGIS Portal Seite, direkt von einer LR Homepage, ...). Wo sollte der Disclaimer hin?

Sollte der Disclaimer beim Laden der Karte angezeigt werden, kann dazu die ``custom.js`` genutzt werden. 

``` javascript

(function () {
    webgis.events.on('viewer-map-created', function (channel, sender, map) {

        webgis.confirm({
            title: "Disclaimer",
            height: "300px",
            message: "Möchtest du diese Karte wirklich nutzen? ...",
            okText: "Ok",
            cancelText: "Nein, danke!",
            onCancel: function () {
                document.location = "https://localhost"
            }
        });
    });
});

```

Sobald die Karte geladen wurde (Event ```viewer-map-created``) kann ein Dialog angezeigt werden. Diesen muss der Anwender dann bestätigen, sonst wird weitergeleitet:

![image](https://user-images.githubusercontent.com/62870568/235605241-a2c00623-02ca-4ba0-965b-c00dedea5320.png)

**Comment by gis-eni** [comment]
``` Javascript

if(webgis.localStorage.get('gelesen')!=='1') {
   // disclaimer anzeigen

   webgis.confirm({
            title: "Disclaimer",
            height: "300px",
            message: "Möchtest du diese Karte wirklich nutzen? ...",
            okText: "Ok, nicht mehr anzeigen",
            cancelText: "Nein, danke!",
            onOk: function() {
                 webgis.localStorage.set('gelesen', '1');
            },
            onCancel: function () {
                document.location = "https://localhost"
            }
        });
}

```

**Reply by marko-jer** [reply]
```js
if(webgis.localStorage.get('gelesen')!=='1') {
   // disclaimer anzeigen

   webgis.events.on('viewer-map-created', function (channel, sender, map) {


	   webgis.confirm({
				title: "Disclaimer",
				height: "300px",
				message: "Möchtest du diese Karte wirklich nutzen? ...",
				okText: "Ok, nicht mehr anzeigen",
				cancelText: "Nein, danke!",
				onOk: function() {
					 webgis.localStorage.set('gelesen', '1');
				},
				onCancel: function () {
					document.location = "https://wichtig.at"
				}
			});
   });
};
```

 ... die Zeile mit `webgis.events.on(...` wird auch benötigt, damit es funktioniert

**Reply by marko-jer** [reply]
Wenn man die Message auch formatieren könnte (Zeilenumbruch, Links..) , wäre es leichter lesbar und informativer.
